src/runtime.js

```diff
@@ -151,6 +151,7 @@ let tornado = {
    */
   exists(val, placeholderNode, bodies, context) {
     if (this.util.isPromise(val)) {
+      placeholderNode = this.insertPendingBody(placeholderNode, bodies.pending, context) || placeholderNode;
```

Comment:
does this need to live in runtime? Is there a general (DRYer) way to handle `{:else}` and `{:pending}` etc. If we add a few more bodies, we will have to litter these everywhere.